agent/dns_node_lookup_test.go

```diff
@@ -18,97 +18,101 @@ func TestDNS_NodeLookup(t *testing.T) {
 	}
 
 	t.Parallel()
-	a := NewTestAgent(t, "")
-	defer a.Shutdown()
-	testrpc.WaitForLeader(t, a.RPC, "dc1")
-
-	// Register node
-	args := &structs.RegisterRequest{
-		Datacenter: "dc1",
-		Node:       "foo",
-		Address:    "127.0.0.1",
-		TaggedAddresses: map[string]string{
-			"wan": "127.0.0.2",
-		},
-		NodeMeta: map[string]string{
-			"key": "value",
-		},
+	for name, experimentsHCL := range versionHCL {
+		t.Run(name, func(t *testing.T) {
+			a := NewTestAgent(t, experimentsHCL)
+			defer a.Shutdown()
+			testrpc.WaitForLeader(t, a.RPC, "dc1")
```

Comment:
Just a random thought for when we write the V2 catalog API tests:
We won't be able to use `testrpc.WaitForLeader`  because it uses the V1 ListNodes catalog endpoint. We'll have to switch to the [V2 version](https://github.com/hashicorp/consul-enterprise/pull/8025/files#diff-4a1450cf93025661a56d461c49be372d3be926137605fac1986938ae279419b8R50-R80) that uses the operator endpoint or something similar. It will be mostly a drop-in substitute except in tests that enable ACLs, like the DNS Token ones.